api/utils.py

```diff
@@ -162,3 +163,34 @@ def get_folio_regex(language, file_name, folio) -> str:
             start_folio = file_name + "_" + folio + ":"
 
     return start_folio
+
+def add_source_information(filename,query_result):
```

Comment:
This would be good for ALL languages and it can be made simpler:
"the source of this text is <<insertSource>>. Click on ... etc. instead of the if statements.

Reply:
Yes you are right! This deserves some more attention.